pallets/asset-manager/src/lib.rs

```diff
@@ -394,6 +431,74 @@ pub mod pallet {
 		pub fn account_id() -> T::AccountId {
 			T::PalletId::get().into_account()
 		}
+
+		/// Strip AccountId from the multilocation.
+		pub(crate) fn extract_multilocation(
+			location: &MultiLocation,
+		) -> Option<<T::AssetConfig as AssetConfig<T>>::AssetLocation> {
+			let MultiLocation { parents, interior } = location;
+			// Currently, we only support X3 at most.
+			let striped_location = match interior {
+				// Send tokens back to relaychain.
+				Junctions::X1(Junction::AccountId32 { .. }) | Junctions::Here => {
+					MultiLocation::parent()
+				}
+				// Send native tokens to sibling chain.
+				Junctions::X2(Junction::Parachain(_para_id), Junction::AccountId32 { .. }) => {
+					// MultiLocation::new(*parents, Junctions::X1(Junction::Parachain(*para_id)))
+					interior.clone().split_last().0.into_exterior(*parents)
+				}
+				// Send tokens to specified pallet on sibling chain.
+				Junctions::X3(
+					Junction::Parachain(_para_id),
+					Junction::PalletInstance(_pallet_idx),
+					Junction::AccountId32 { .. },
+				) => interior.clone().split_last().0.into_exterior(*parents),
+				// Send tokens with specified token symbol to sibling chain.
+				Junctions::X3(
+					Junction::Parachain(_para_id),
+					Junction::GeneralKey(_key),
+					Junction::AccountId32 { .. },
+				) => interior.clone().split_last().0.into_exterior(*parents),
+				// For now, we don't support X4 or longer Junctions
+				_ => return None,
```

Comment:
I think the only locations we need is `(1, X2(Parachain, AccountId))` and `(1, X1(AccountId))`